### PR TITLE
Mona: Tasks from 151 to 160

### PR DESCRIPTION
**Pull request by hsp-mona**
Mona: Tasks from 151 to 160

### CODE REVIEW

File: src/main/java/am/hsp/apps/seua/Base/TaskBase.java [hunk 1]
```diff
@@ -124,11 +124,27 @@ protected boolean isOdd(int number) {
...
+		
+		return result;
+	}
+		
 	/**
 	 * Get the quotient of given two numbers
 	 */
 	protected double quotientOfTwoNumbers(int number1, int number2) {
-		return ((double) number2 / number1);
+		return ((double) number1 / (double) number2);
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Base/TaskBase.java]: Issue is written in your previous PR

File: src/main/java/am/hsp/apps/seua/Base/TaskNaturalNumbers.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package main.java.am.hsp.apps.seua.Base;
+/**
+ * Class will contain methods for natural numbers.
+ * @author mona
+ *
+ */
+
+public abstract class TaskNaturalNumbers extends TaskBase {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Base/TaskNaturalNumbers.java]: Class name should be singular (as it already done for other classes like TaskDigit).

File: src/main/java/am/hsp/apps/seua/Base/TaskNaturalNumbers.java [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+	 */
+	protected boolean isDivisor(int number, int divisor) {
+		return(number % divisor == 0);
+	}
+	
+	/**
+	 * Checks is the number  not divisor for given number
+	 */
+	protected boolean isNotDivisor(int number, int divisor) {
+		return(!isDivisor(number, divisor));
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Base/TaskNaturalNumbers.java]: braces are not required after return if there is only one check condition.

File: src/main/java/am/hsp/apps/seua/Task_151.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+		System.out.println("Sum of divisors is equal to " + getSumOfDivisors(number));
+	}
+	
+	/**
+	 * get sum of divisors of given number
+	 */
+	private int getSumOfDivisors(int number) {
+		int divisor = 1;
+		int sum = 0;
+		while (divisor <= number) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_151.java]: I guess there is no need to continue the loop until divisor will be equal to number. The max divisor should be "number / 2".
If you are doing this to include also the "number" as its divisor, you just need to add the number to "sum" after the loop...

File: src/main/java/am/hsp/apps/seua/Task_152.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+		System.out.println("Product of divisors is equal to " + getProductOfDivisors(number));
+	}
+	
+	/**
+	 * get product of divisors of given number
+	 */
+	private int getProductOfDivisors(int number) {
+		int divisor = 1;
+		int product = 1;
+		while (divisor <= number) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_152.java]: the same issue as in previous task...

File: src/main/java/am/hsp/apps/seua/Task_153.java [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+				"Sum of divisors by 2 balance is equal to " + getSumOfDivisorsByGivenBalance(number, BALLANCE_VALUE));
+	}
+
+	/**
+	 * get sum of divisors of given number
+	 */
+	private int getSumOfDivisorsByGivenBalance(int number, int balance) {
+		int divisor = 1;
+		int sum = 0;
+		while (divisor <= number) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_153.java]: I guess the max divisor is "number - balance"

File: src/main/java/am/hsp/apps/seua/Task_154.java [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+				"Product of divisors by 3 balance is equal to " + getProductOfDivisorsByGivenBalance(number, BALLANCE_VALUE));
+	}
+
+	/**
+	 * get sum of divisors of given number
+	 */
+	private int getProductOfDivisorsByGivenBalance(int number, int balance) {
+		int divisor = 1;
+		int product = 1;
+		while (divisor <= number) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_154.java]: the same issue as in previous task

File: src/main/java/am/hsp/apps/seua/Task_155.java [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Get the sum of two-digit numbers which is multiple of 3.
+ * 
+ * @author mona
+ *
+ */
+
+public class Task_155 extends TaskNaturalNumbers {
+	private static final int LOWER_BORDER_VALUE = 10;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_155.java]: Keep it simple. MIN_VALUE = 10 ; MAX_VALUE =  99; and use <= in the while loop

File: src/main/java/am/hsp/apps/seua/Task_155.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+	@Override
+	public void solve() {
+		System.out.println(
+				"Sum of two-digit numbers which is multiple of 3 is equal to " + getSumOfMultiplesofGivenNumber());
+	}
+
+	/**
+	 * get sum of two-digit numbers which is multiple of given number
+	 */
+	private int getSumOfMultiplesofGivenNumber() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_155.java]: To have a generic method you need to pass dynamic data as its parameters.The method name could be simplified also:
getSumOfMultiples(int from, int to, int divisor)

File: src/main/java/am/hsp/apps/seua/Task_156.java [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+	@Override
+	public void solve() {
+		System.out.println("Product of two-digit numbers which is multiple of 3 and 5 is equal to "
+				+ getProductOfMultiplesofGivenNumbers());
+	}
+
+	/**
+	 * get product of two-digit numbers which is multiple of given number
+	 */
+	private long getProductOfMultiplesofGivenNumbers() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_156.java]: same issue as in previous task

File: src/main/java/am/hsp/apps/seua/Task_156.java [hunk 11]
```diff
@@ -0,0 +1,43 @@
...
+
+/**
+ * Get the product of two-digit numbers which is multiple of 3 and 5.
+ * 
+ * @author mona
+ *
+ */
+
+public class Task_156 extends TaskNaturalNumbers {
+	private static final int LOWER_BORDER_VALUE = 10;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_156.java]: See issues in the previous task...

File: src/main/java/am/hsp/apps/seua/Task_157.java [hunk 12]
```diff
@@ -0,0 +1,41 @@
...
+	@Override
+	public void solve() {
+		System.out.println("Sum of three-digit numbers which is non-multiple of 5 is equal to "
+			+ getSumOfNonMultiplesofGivenNumber());
+	}
+		
+	/**
+	 * get sum of three-digit numbers which is non-multiple of given number
+	 */
+	private int getSumOfNonMultiplesofGivenNumber() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_157.java]: same issue as in previous task...

File: src/main/java/am/hsp/apps/seua/Task_157.java [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+import main.java.am.hsp.apps.seua.Base.TaskNaturalNumbers;
+
+/**
+ * Get sum of three-digit numbers which is non-multiple of 5. 
+ * @author mona
+ *
+ */
+
+public class Task_157 extends TaskNaturalNumbers {
+	private static final int LOWER_BORDER_VALUE = 100;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_157.java]: same...

File: src/main/java/am/hsp/apps/seua/Task_158.java [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+import main.java.am.hsp.apps.seua.Base.TaskNaturalNumbers;
+
+/**
+ * Get product of three-digit numbers which is non-multiple of 2 and 3.
+ * @author mona
+ *
+ */
+
+public class Task_158 extends TaskNaturalNumbers {
+	private static final int LOWER_BORDER_VALUE = 100;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_158.java]: same...

File: src/main/java/am/hsp/apps/seua/Task_158.java [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+	@Override
+	public void solve() {
+		System.out.println("Product of three-digit numbers which is non-multiple of 2 and 3 is equal to "
+				+ getProductOfNonMultiplesofGivenNumbers());
+	}
+
+	/**
+	 * get product of three-digit numbers which is non-multiple of given numbers
+	 */
+	private long getProductOfNonMultiplesofGivenNumbers() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_158.java]: same

File: src/main/java/am/hsp/apps/seua/Task_159.java [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+/**
+ * Get the product of three-digit numbers which are divisors of 3 by 1 balance
+ * and divisors of 4 by 2 balance
+ * 
+ * @author mona
+ *
+ */
+
+public class Task_159 extends TaskNaturalNumbers {
+	private static final int LOWER_BORDER_VALUE = 100;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_159.java]: same

File: src/main/java/am/hsp/apps/seua/Task_159.java [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+	public void solve() {
+		System.out.println("Product of divisors of 3 by 1 balance and divisors of 4 by 2 balance  is equal to "
+				+ getProductOfMultiplesByGivenBalances());
+	}
+
+	/**
+	 * get product of three-digit numbers which is multiple of given numbers by
+	 * given balances
+	 */
+	private long getProductOfMultiplesByGivenBalances() {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_159.java]: same. Also please remind me to discuss this task in our lesson...

File: src/main/java/am/hsp/apps/seua/Task_160.java [hunk 18]
```diff
@@ -0,0 +1,34 @@
+package main.java.am.hsp.apps.seua;
+
+import main.java.am.hsp.apps.seua.Base.TaskNaturalNumbers;
+
+public class Task_160 extends TaskNaturalNumbers {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_160.java]: Comment required...

File: src/main/java/am/hsp/apps/seua/Task_160.java [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+	private static final int LOWER_BORDER_VALUE = 100;
+	private static final int OVER_BORDER_VALUE = 1000;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_160.java]: same...

File: src/main/java/am/hsp/apps/seua/Task_160.java [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+
+	public static void main(String[] args) {
+		Task_160 task = new Task_160();
+		task.solve();
+	}
+
+	@Override
+	public void solve() {
+		System.out.println(
+				"Minimum tree-digit number whose 26 multipled is rooted is equal to " + getMinNMultipledRootedNumber());
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_160.java]: I guess you implemented some kind of Mix of Task 160 and 161 :) please check!

File: src/main/java/am/hsp/apps/seua/Task_160.java [hunk 21]
```diff
@@ -0,0 +1,34 @@
...
+	}
+
+	private int getMinNMultipledRootedNumber() {
+		int result = 0;
+		for (int number = LOWER_BORDER_VALUE; number < OVER_BORDER_VALUE; number++) {
+			for (int root = 1; root < MULTIPLE_NUMBER * number; root++) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_160.java]: You need to simplify this method. You need to create a new method which will check whether given number has a square root or not. **boolean hasSquareRoot(in number)**

File: src/main/java/am/hsp/apps/seua/Base/TaskNaturalNumbers.java (hunk 3 above)
Comment (hsp-mona) [reply src/main/java/am/hsp/apps/seua/Base/TaskNaturalNumbers.java]: But I've observed number case too, cause I think that number is divisor itself too.

File: src/main/java/am/hsp/apps/seua/Task_151.java (hunk 4 above)
Comment (hsp-mona) [reply src/main/java/am/hsp/apps/seua/Task_151.java]: But I've observed number case too, cause I think that number is divisor itself too.

File: src/main/java/am/hsp/apps/seua/Task_152.java (hunk 5 above)
Comment (hsp-mona) [reply src/main/java/am/hsp/apps/seua/Task_152.java]: the same comment as in previous task ... :)

File: src/main/java/am/hsp/apps/seua/Task_160.java [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+				if (MULTIPLE_NUMBER * number == pow(root, 2)) {
+					result = number;
+					number = 1000;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_160.java]: IT is not clear why you are setting 1000 to number variable? Based on your task you need to return the lowest number for which required condition is matched. So why are you continuing your "for" loops even when desired number is already found?

File: src/main/java/am/hsp/apps/seua/Base/TaskNaturalNumber.java [hunk 23]
```diff
@@ -0,0 +1,139 @@
...
+	 * @param to
+	 * @param multiplier
+	 * @return
+	 */
+	protected int getMinNMultipledRootedNumber(int from, int to, int multiplier) {
+		int result = 0;
+		for (int number = from; number <= to; number++) {
+			if (hasNumberNaturalRoot(multiplier * number)) {
+				result = number;
+				number = to;
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Base/TaskNaturalNumber.java]: Please take a look at "break/continue" operations So when you write "break" - your loop will not be continued anymore, and there will be no need to do number = to

Anyway if you found the result here, there is no need to continue, just return a number and that's all :)

File: src/main/java/am/hsp/apps/seua/Task_151.java [hunk 24]
```diff
@@ -0,0 +1,41 @@
...
+	public void solve() {
+		int number = getUserInput("Please enter the number. ");
+		
+		System.out.println("Sum of divisors is equal to " + getSumOfDivisors(number));
+	}
+	
+	/**
+	 * get sum of divisors of given number
+	 */
+	private int getSumOfDivisors(int number) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_151.java]: I guess this is the same as getSumOfMultiples method in one of base classes. So DRY violation here. Please fix.

File: src/main/java/am/hsp/apps/seua/Task_152.java [hunk 25]
```diff
@@ -0,0 +1,41 @@
...
+	public void solve() {
+		int number = getUserInput("Please enter the number. ");
+
+		System.out.println("Product of divisors is equal to " + getProductOfDivisors(number));
+	}
+	
+	/**
+	 * get product of divisors of given number
+	 */
+	private int getProductOfDivisors(int number) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_152.java]: If you have decided to put getSumOfMultiples method in the Base class, so in that case you need to put this method also in the base class.

File: src/main/java/am/hsp/apps/seua/Task_153.java [hunk 26]
```diff
@@ -0,0 +1,44 @@
...
+		int number = getUserInput("Please enter the number. ");
+
+		System.out.println(
+				"Sum of divisors by 2 balance is equal to " + getSumOfDivisorsByGivenBalance(number, BALLANCE_VALUE));
+	}
+
+	/**
+	 * get sum of divisors of given number
+	 */
+	private int getSumOfDivisorsByGivenBalance(int number, int balance) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_153.java]: I guess you have similar method in your base class

File: src/main/java/am/hsp/apps/seua/Task_154.java [hunk 27]
```diff
@@ -0,0 +1,45 @@
...
+		int number = getUserInput("Please enter the number. ");
+
+		System.out.println(
+				"Product of divisors by 3 balance is equal to " + getProductOfDivisorsByGivenBalance(number, BALLANCE_VALUE));
+	}
+
+	/**
+	 * get sum of divisors of given number
+	 */
+	private int getProductOfDivisorsByGivenBalance(int number, int balance) {
```
Comment (hsp-harut) [inline src/main/java/am/hsp/apps/seua/Task_154.java]: same as for above method...

**Pull request merged**